geode-core/src/main/java/org/apache/geode/cache/query/internal/DefaultQuery.java

```diff
@@ -317,33 +297,16 @@ public Object execute(Object[] params) throws FunctionDomainException, TypeMisma
       }
       return result;
     } catch (QueryExecutionCanceledException ignore) {
-      return reinterpretQueryExecutionCanceledException();
+      return context.reinterpretQueryExecutionCanceledException();
```

Comment:
👍 making reinterpretation the responsibility of the query execution context
👎 failing to pass the exception we just caught…

Is the exception we get from `ExecutionContext.getQueryCanceledException()` different from the one we just caught??